openresty/Dockerfile-alpine

```
@@ -0,0 +1,253 @@
+# Specify the LUAJIT_LIB-s that are already installed in the base image.
+# use the includes from the bundled version. This is where LUA JIT VERSION is important. Because it differs from release to release (probably).
+RUN set -eux; \
+    git clone -b master --depth 1 https://github.com/SpiderLabs/ModSecurity-nginx.git; \
+    wget --quiet https://openresty.org/download/openresty-"${OPENRESTY_VERSION}".tar.gz; \
+    tar -xzf openresty-"${OPENRESTY_VERSION}".tar.gz; \
+    cd ./openresty-"${OPENRESTY_VERSION}"/bundle/nginx-${NGINX_VERSION}; \
+    export LUAJIT_LIB="/usr/local/openresty/luajit/lib/"; \
+    export LUAJIT_INC="../LuaJIT-${LUA_JIT_VERSION}/src/"; \
+    COMPILEOPTIONS=$(openresty -V 2>&1| grep -i "arguments"|cut -d ":" -f2-); \
+    eval ./configure $COMPILEOPTIONS --add-dynamic-module=../../../ModSecurity-nginx; \
+    make modules; \
+    cp objs/ngx_http_modsecurity_module.so /usr/local/openresty/nginx/modules/; \ 
+    mkdir /etc/modsecurity.d; \
+    wget --quiet https://raw.githubusercontent.com/SpiderLabs/ModSecurity/v3/master/modsecurity.conf-recommended \
+        -O /etc/modsecurity.d/modsecurity.conf; \
+    wget --quiet https://raw.githubusercontent.com/SpiderLabs/ModSecurity/v3/master/unicode.mapping \
+        -O /etc/modsecurity.d/unicode.mapping
+
+
+# Generate self-signed certificates (if needed)
+RUN mkdir -p /usr/share/TLS
+COPY openresty/openssl.conf /usr/share/TLS
+RUN /usr/local/openresty/openssl/bin/openssl req -x509 -days 365 -new \
+    -config /usr/share/TLS/openssl.conf \
+    -keyout /usr/share/TLS/server.key \
+    -out /usr/share/TLS/server.crt
+
+# Generate/Download Diffie-Hellman parameter files
+RUN /usr/local/openresty/openssl/bin/openssl dhparam -out /usr/share/TLS/dhparam-1024.pem 1024
+RUN curl -sSL https://ssl-config.mozilla.org/ffdhe2048.txt -o /usr/share/TLS/dhparam-2048.pem
+RUN curl -sSL https://ssl-config.mozilla.org/ffdhe4096.txt -o /usr/share/TLS/dhparam-4096.pem
+
+FROM openresty/openresty:${OPENRESTY_VERSION}-alpine-fat as crs_release
+
+ARG CRS_RELEASE
+
+# hadolint ignore=DL3008,SC2016
+RUN set -eux; \
+    apk add --no-cache \
+    ca-certificates \
+    curl \
+    gnupg; \
+    mkdir /opt/owasp-crs; \
+    curl -SL https://github.com/coreruleset/coreruleset/archive/v${CRS_RELEASE}.tar.gz -o v${CRS_RELEASE}.tar.gz; \
+    curl -SL https://github.com/coreruleset/coreruleset/releases/download/v${CRS_RELEASE}/coreruleset-${CRS_RELEASE}.tar.gz.asc -o coreruleset-${CRS_RELEASE}.tar.gz.asc; \
+    gpg --fetch-key https://coreruleset.org/security.asc; \
+    gpg --verify coreruleset-${CRS_RELEASE}.tar.gz.asc v${CRS_RELEASE}.tar.gz; \
+    tar -zxf v${CRS_RELEASE}.tar.gz --strip-components=1 -C /opt/owasp-crs; \
+    rm -f v${CRS_RELEASE}.tar.gz coreruleset-${CRS_RELEASE}.tar.gz.asc; \
+    mv -v /opt/owasp-crs/crs-setup.conf.example /opt/owasp-crs/crs-setup.conf
+
+FROM openresty/openresty:${OPENRESTY_VERSION}-alpine-fat
+
+ARG MODSEC_VERSION=3.0.9
```

Use value from top-level ARG:
```suggestion
ARG MODSEC_VERSION
```